src/iop/cacorrect.c

```diff
@@ -1493,8 +1493,9 @@ void reload_defaults(dt_iop_module_t *module)
   // we might be called from presets update infrastructure => there is no image
   if(!module->dev) goto end;
 
+  dt_image_t *img = &module->dev->image_storage;
   // can't be switched on for non-raw or x-trans images:
-  if(dt_image_is_raw(&module->dev->image_storage) && (module->dev->image_storage.buf_dsc.filters != 9u))
+  if(dt_image_is_raw(img) && (img->buf_dsc.filters != 9u) && !dt_image_is_monochrome(img))
```

Comment:
Resetting this module apparently enables it, and it can't be disabled back

Reply:
Interesting, I'll fix this.

Reply:
To be honest, i'm not sure what the fix should be.
At least, i think, you should edit `commit_params()`, and add
`if(dt_image_is_monochrome(img)) {piece->enabled = false; return;}`

Reply:
> However, after resetting, (this was the case before), this adds a history item, and it doesn't say (off); is that a bug?

This is consistent with all the previous behavior.
In the current context it highlights just how ad-hoc hack all these "this module is not supported for this image" implementations are.

I guess you should modify `_reset_label_callback()` to *not* enable module *if* it is *not* enabled by default AND the toggle button is *hidden*. At least that would make sense to me. But i have no clue whatsoever what will get broken by that.